main.go

```diff
@@ -36,15 +36,43 @@ func connectionChecker(peer smtpd.Peer) error {
 	return smtpd.Error{Code: 421, Message: "Denied"}
 }
 
+func addrAllowed(addr string, allowedAddrs []string) bool {
+	if allowedAddrs == nil {
+		return true
+	}
+
+	domidx := strings.LastIndex(addr, "@")
```

Comment:
Since addr is the sender email it shoud be valid that the address does not contain a domain and @ yet.

Reply:
Per [below](https://github.com/decke/smtprelay/pull/9#issuecomment-778836237), I was unaware that "local" addresses could reach smtprelay. Updated (see below).